### PR TITLE
Restoring disable-runtime-api for test chains

### PR DESCRIPTION
**Pull request by bolajahmad**
Adds disable-runtime-api back to rococo and westend. Previously removed.

Partly closes, #1621 . Kusams and Polkadot will be updated on the [runtimes codebase](https://github.com/bolajahmad/runtimes) in a seperate PR.

### CODE REVIEW

File: polkadot/runtime/westend/src/lib.rs [hunk 1]
```diff
@@ -191,15 +192,30 @@ parameter_types! {
...
+	type RuntimeTask = RuntimeTask;
+	type PalletInfo = PalletInfo;
+	type OnNewAccount = ();
+	type OnKilledAccount = ();
+	type OnSetCode = ();
+	type SingleBlockMigrations = ();
+	type MultiBlockMigrator = ();
+	type PreInherents = ();
+	type PostInherents = ();
+	type PostTransactions = ();
```
Comment (bkchr) [inline polkadot/runtime/westend/src/lib.rs:209]:
```suggestion
```

File: polkadot/runtime/rococo/src/lib.rs [hunk 2]
```diff
@@ -199,6 +201,21 @@ parameter_types! {
...
+	type RuntimeTask = RuntimeTask;
+	type PalletInfo = PalletInfo;
+	type OnNewAccount = ();
+	type OnKilledAccount = ();
+	type OnSetCode = ();
+	type SingleBlockMigrations = ();
+	type MultiBlockMigrator = ();
+	type PreInherents = ();
+	type PostInherents = ();
+	type PostTransactions = ();
```
Comment (bkchr) [inline polkadot/runtime/rococo/src/lib.rs:218]:
```suggestion
```

File: polkadot/runtime/rococo/Cargo.toml [hunk 3]
```diff
@@ -323,6 +330,8 @@ try-runtime = [
 	"sp-runtime/try-runtime",
 ]
 
+disable-runtime-api = []
```
Comment (bkchr) [inline polkadot/runtime/rococo/Cargo.toml:333]: Some comment would be nice.

Comment (bolajahmad) [reply polkadot/runtime/rococo/Cargo.toml:333]: That makes sense and I will add descriptive comments.
One thing I'd like to confirm though, is this still required by Cumulus to access certain runtime(s)?

File: polkadot/runtime/westend/src/lib.rs (hunk 1 above)
Comment (bolajahmad) [reply polkadot/runtime/westend/src/lib.rs:209]: Screenshot 2024-05-03 at 19 20 49

The issue with removing these changes is that the runtime demands that they are implemented. These types should not be needed but I am not sure why the compiler shows this error. Are there any recommendations from your end @bkchr

File: polkadot/runtime/rococo/src/lib.rs (hunk 2 above)
Comment (bolajahmad) [reply polkadot/runtime/rococo/src/lib.rs:218]: Screenshot 2024-05-03 at 19 24 43

Just like the previous comment, the types should not be needed but I am not sure why the compiler shows this error. Are there any recommendations from your end @bkchr

Comment (bkchr) [reply polkadot/runtime/rococo/src/lib.rs:218]: Did you try `cargo build`? `rust-analyzer` is may just confused.

Comment (bolajahmad) [reply polkadot/runtime/rococo/src/lib.rs]: It does work, rust-analyzer was truly the problem here. I got it to build without the additional types. Thanks @bkchr

File: polkadot/runtime/rococo/Cargo.toml (hunk 3 above)
Comment (bolajahmad) [reply polkadot/runtime/rococo/Cargo.toml:335]: Added some comments, let me know if they're descriptive enough.

File: polkadot/runtime/rococo/Cargo.toml [hunk 4]
```diff
@@ -324,6 +331,9 @@ try-runtime = [
 	"sp-runtime/try-runtime",
 ]
 
+# When enabled, the runtime API will not be built.
```
Comment (bkchr) [inline polkadot/runtime/rococo/Cargo.toml:334]:
```suggestion
# If enabled, the runtime apis will not be exposed by the runtime. 
```

File: polkadot/runtime/westend/Cargo.toml [hunk 5]
```diff
@@ -338,6 +346,9 @@ try-runtime = [
 	"sp-runtime/try-runtime",
 ]
 
+# When enabled, the runtime API will not be built.
```
Comment (bkchr) [inline polkadot/runtime/westend/Cargo.toml:349]:
```suggestion
# If enabled, the runtime apis will not be exposed by the runtime.
```

**Review by bkchr — COMMENTED** [review]
Can you please revert the unrelated changes on your pr? Otherwise it looks good.

File: polkadot/runtime/westend/src/lib.rs (hunk 1 above)
Comment (bolajahmad) [reply polkadot/runtime/westend/src/lib.rs]: These have been removed, the errors were from rust-analyzer and project still builds

**Comment by bkchr** [comment]
@bolajahmad could you please revert the changes that are not related to your pr?

**Comment by bolajahmad** [comment]
> @bolajahmad could you please revert the changes that are not related to your pr?

Hi @bkchr, i tried to cleanup the branch and revert unrelated changes to my PR. I've removed the changes that seemed weird and it should be good. I'm sorry to bother you when you're reviewing my changes but I can't find anymore unrelated changes in this PR. I'll appreciate if you mention something specific you think I need to remove/revert too🙏🏼

**Comment by bkchr** [comment]
https://github.com/paritytech/polkadot-sdk/pull/4325/files#diff-5264d812f7aadfc486efffd3322b11c8dee5e8ddf2e77e0a9e70b573f3d2afddR91 here for example where you import `AccountIdLookup` which was not there before.

Or the formatting changes [here](https://github.com/paritytech/polkadot-sdk/pull/4325/files#diff-da220c6c064cc047fef5d3b4a594ae30b8bbf48a29b740742bc9858285c7ba57R902)

**Comment by bkchr** [comment]
The issue was closed as not needed anymore, so we can do the same with this pr.

**Pull request closed without merging**